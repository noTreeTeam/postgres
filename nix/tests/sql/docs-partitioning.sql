-- testing sql found in https://supabase.com/docs/guides/database/partitioning

create table sales (
    id bigint generated by default as identity,
    order_date date not null,
    customer_id bigint,
    amount bigint,
    primary key (order_date, id)
)
partition by range (order_date);

create table sales_2000_01
	partition of sales
  for values from ('2000-01-01') to ('2000-02-01');

create table sales_2000_02
	partition of sales
	for values from ('2000-02-01') to ('2000-03-01');

insert into sales (order_date, customer_id, amount) values
  ('2000-01-15', 1, 100),
  ('2000-01-20', 2, 200),
  ('2000-02-10', 3, 150),
  ('2000-02-25', 4, 300);

select * from sales where order_date >= '2000-01-01' and order_date < '2000-03-01';

select * from sales_2000_02;

drop table sales cascade;

create table customers (
    id bigint generated by default as identity,
    name text,
    country text,
    primary key (country, id)
)
partition by list(country);

create table customers_americas
	partition of customers
	for values in ('US', 'CANADA');

create table customers_asia
	partition of customers
  for values in ('INDIA', 'CHINA', 'JAPAN');

insert into customers (name, country) values
  ('John Doe', 'US'),
  ('Jane Smith', 'CANADA'),
  ('Li Wei', 'CHINA'),
  ('Priya Patel', 'INDIA'),
  ('Yuki Tanaka', 'JAPAN');

select * from customers where country in ('US', 'CANADA');

select * from customers_asia;

drop table customers cascade;

create table products (
    id bigint generated by default as identity,
    name text,
    category text,
    price bigint
)
partition by hash (id);

create table products_one
	partition of products
  for values with (modulus 2, remainder 1);

create table products_two
	partition of products
  for values with (modulus 2, remainder 0);

insert into products (name, category, price) values
  ('Laptop', 'Electronics', 999),
  ('Phone', 'Electronics', 599),
  ('Book', 'Education', 29),
  ('Chair', 'Furniture', 199);

select * from products where category = 'Electronics';

select count(*) from products_one;

select count(*) from products_two;

drop table products cascade; 
