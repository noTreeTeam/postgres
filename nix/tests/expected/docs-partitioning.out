-- testing sql found in https://supabase.com/docs/guides/database/partitioning
create table sales (
    id bigint generated by default as identity,
    order_date date not null,
    customer_id bigint,
    amount bigint,
    primary key (order_date, id)
)
partition by range (order_date);
create table sales_2000_01
	partition of sales
  for values from ('2000-01-01') to ('2000-02-01');
create table sales_2000_02
	partition of sales
	for values from ('2000-02-01') to ('2000-03-01');
insert into sales (order_date, customer_id, amount) values
  ('2000-01-15', 1, 100),
  ('2000-01-20', 2, 200),
  ('2000-02-10', 3, 150),
  ('2000-02-25', 4, 300);
select * from sales where order_date >= '2000-01-01' and order_date < '2000-03-01';
 id | order_date | customer_id | amount 
----+------------+-------------+--------
  1 | 01-15-2000 |           1 |    100
  2 | 01-20-2000 |           2 |    200
  3 | 02-10-2000 |           3 |    150
  4 | 02-25-2000 |           4 |    300
(4 rows)

select * from sales_2000_02;
 id | order_date | customer_id | amount 
----+------------+-------------+--------
  3 | 02-10-2000 |           3 |    150
  4 | 02-25-2000 |           4 |    300
(2 rows)

drop table sales cascade;
create table customers (
    id bigint generated by default as identity,
    name text,
    country text,
    primary key (country, id)
)
partition by list(country);
create table customers_americas
	partition of customers
	for values in ('US', 'CANADA');
create table customers_asia
	partition of customers
  for values in ('INDIA', 'CHINA', 'JAPAN');
insert into customers (name, country) values
  ('John Doe', 'US'),
  ('Jane Smith', 'CANADA'),
  ('Li Wei', 'CHINA'),
  ('Priya Patel', 'INDIA'),
  ('Yuki Tanaka', 'JAPAN');
select * from customers where country in ('US', 'CANADA');
 id |    name    | country 
----+------------+---------
  1 | John Doe   | US
  2 | Jane Smith | CANADA
(2 rows)

select * from customers_asia;
 id |    name     | country 
----+-------------+---------
  3 | Li Wei      | CHINA
  4 | Priya Patel | INDIA
  5 | Yuki Tanaka | JAPAN
(3 rows)

drop table customers cascade;
create table products (
    id bigint generated by default as identity,
    name text,
    category text,
    price bigint
)
partition by hash (id);
create table products_one
	partition of products
  for values with (modulus 2, remainder 1);
create table products_two
	partition of products
  for values with (modulus 2, remainder 0);
insert into products (name, category, price) values
  ('Laptop', 'Electronics', 999),
  ('Phone', 'Electronics', 599),
  ('Book', 'Education', 29),
  ('Chair', 'Furniture', 199);
select * from products where category = 'Electronics';
 id |  name  |  category   | price 
----+--------+-------------+-------
  1 | Laptop | Electronics |   999
  2 | Phone  | Electronics |   599
(2 rows)

select count(*) from products_one;
 count 
-------
     2
(1 row)

select count(*) from products_two;
 count 
-------
     2
(1 row)

drop table products cascade; 
